### PR TITLE
Move Blockchain.Initialize

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2487

### CODE REVIEW

**Comment by ZhangTao1596** [comment]
If so, all plugins will miss genesis block.

**Comment by nicolegys** [comment]
As @ZhangTao1596 said, `OnPersist` will be called before `OnSystemLoaded`, which will cause some problems. 
e.g.
image

**Comment by shargon** [comment]
@devhawk any other solution?

**Comment by Liaojinghui** [comment]
Why not move `GenesisBlock` to the dbft plugin?

**Comment by devhawk** [comment]
> @devhawk any other solution? load first storage plugins?

Let me take a look at the code today and get back to you.

File: src/neo/NeoSystem.cs [hunk 1]
```diff
@@ -116,9 +116,9 @@ public NeoSystem(ProtocolSettings settings, string storageEngine = null, string
             this.LocalNode = ActorSystem.ActorOf(Network.P2P.LocalNode.Props(this));
             this.TaskManager = ActorSystem.ActorOf(Network.P2P.TaskManager.Props(this));
             this.TxRouter = ActorSystem.ActorOf(TransactionRouter.Props(this));
+            Blockchain.Ask(new Blockchain.Initialize()).Wait();
```
Comment (devhawk) [inline src/neo/NeoSystem.cs:119]: This change does not change the issue identified in #2487

**Comment by devhawk** [comment]
I would recommend moving the following code from the end of `NeoSystem` constructor to the beginning of `StartNode`

``` csharp
foreach (var plugin in Plugin.Plugins)
    plugin.OnSystemLoaded(this);
Blockchain.Ask(new Blockchain.Initialize()).Wait();
```

**Comment by ZhangTao1596** [comment]
@shargon   Harry mean move genesis block tell from NeoSystem constructor outside. So that we can do some extra work between NeoSystem initialization and genesis block.

**Comment by nicolegys** [comment]
> I would recommend moving the following code from the end of `NeoSystem` constructor to the beginning of `StartNode`
> 
> ```cs
> foreach (var plugin in Plugin.Plugins)
>     plugin.OnSystemLoaded(this);
> Blockchain.Ask(new Blockchain.Initialize()).Wait();
> ```

Blockchain should be initialized before `import blocks`, `StartNode` is after `import blocks`. Moving `Blockchain.Ask(new Blockchain.Initialize()).Wait()` to the beginning of `StartNode` may cause problems.

**Comment by devhawk** [comment]
Yeah, @nicolegys is right - simply moving the blockchain init code to StartNode will break the import code. Let me think on this a bit more

**Comment by shargon** [comment]
I will close it until find another solution

**Pull request closed without merging**